channels/chan_console.c

```diff
@@ -401,8 +404,8 @@ static int stop_stream(struct console_pvt *pvt)
 	if (!pvt->streamstate || pvt->thread == AST_PTHREADT_NULL)
 		return 0;
 
-	pthread_cancel(pvt->thread);
```

Comment:
These should still happen to cause the thread to wake up if needed.

Reply:
`pthread_cancel` was the whole reason for the deadlock in the first place, that's why I'd prefer to avoid it. It didn't seem to be necessary since the read eventually expires and it returns on its own.

It might work if `pthread_setcancelstate` is used to disable cancellability in the critical section, but is that the place you're thinking the thread would need to be woken up?